Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -1080,33 +1080,23 @@ Bytecode::CodeGenerationErrorOr<void> ObjectExpression::generate_bytecode(Byteco
 
 Bytecode::CodeGenerationErrorOr<void> ArrayExpression::generate_bytecode(Bytecode::Generator& generator) const
 {
-    Vector<Bytecode::Register> element_regs;
-    for (auto& element : m_elements) {
-        if (element && is<SpreadExpression>(*element)) {
-            return Bytecode::CodeGenerationError {
-                this,
-                "Unimplemented element kind: SpreadExpression"sv,
-            };
-        }
-        element_regs.append(generator.allocate_register());
-    }
-    size_t i = 0;
-    for (auto& element : m_elements) {
```

Comment:
Unfortunately this optimisation is _essential_ for large array literals, we cannot fit infinitely many objects as literals in our bytecode.
again, for this, you can just generate a series of concats instead (if there _are_ any spreads) - pseudocode:
```
// [a, b, c, ...d, e, f]
(a)
store 1
(b)
store 2
(c)
store 3
new array (1..3)
store 4
(d)
[loop code, append to 4]
(e)
append to 4
(f)
append to 4
```